messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -295,12 +281,34 @@ public synchronized Long getReaderBackLog() throws IOException {
       }
       retryCount++;
     }
-    if (lReader.getCurrentFile() != null) {
-      localStreamPendingSize += lReader.getPendingSize();
-      LOG.info("Pending Size of compressed data inside streams_local output location starting from file: " +
-              lReader.getCurrentFile() + " is " + localStreamPendingSize);
+    /**
+     * Reader can be either local stream reader or collector reader at any point of time.
+     * Based on that, the other reader's pending size to be calculated, along with current reader's
+     * pending size. Return sum of both, after adjusting local stream reader's pending size
+     * with compression factor.
+     */
+    if (reader == lReader) {
+      if (lReader.getCurrentFile() != null) { //current reader is local stream reader
+        //get local reader pending size
+        localStreamPendingSize = lReader.getPendingSize(lReader.getCurrentFile());
+        //get collector info from local stream file name and calculate collector pending size
+        String collectorPath = CollectorStreamReader.getCollectorFileName(streamName, lReader.getCurrentFile().getName());
+        collectorStreamPendingSize = cReader.getPendingSize(new Path(collectorPath));
+      }
+    } else if (reader == cReader) { //current reader is collector reader
+      if (cReader.getCurrentFile() != null) {
+        //get collector reader pending size
+        collectorStreamPendingSize = cReader.getPendingSize(cReader.getCurrentFile());
+        //get local reader file path using
+        String localReaderPath = LocalStreamCollectorReader.getDatabusStreamFileName(partitionId.getCollector(), cReader.getCurrentFile().getName());
+        Path localStreamPath = lReader.getFileFromCollectorFileName(localReaderPath);
+        if (localStreamPath != null) {
+          localStreamPendingSize = lReader.getPendingSize(localStreamPath);
```

Comment:
Check this - this might be adding the files that are kept in collector stream (the max files - 6) from both collector stream and local stream.